modelcontextprotocol/server.py

```diff
@@ -457,6 +458,80 @@ def update_assets_tool(
         }
 
 
+@mcp.tool()
+def query_asset_tool(
+    sql: str,
+    connection_qualified_name: str,
+    default_schema: str = None,
+):
+    """
+    Execute a SQL query on a table/view asset.
+
+    This tool enables querying table/view assets on the source similar to 
+    what's available in the insights table. It uses the Atlan query capabilities
+    to execute SQL against connected data sources.
+
+    Note:
+        Only SELECT queries are allowed
+
+    Args:
+        sql (str): The SQL query to execute (SELECT queries only)
+        connection_qualified_name (str): Connection qualified name to use for the query.
+            This is the same parameter used in search_assets_tool.
+            You can find this value by searching for Connection assets using search_assets_tool
+            and looking at the 'qualifiedName' attribute.
+        default_schema (str, optional): Default schema name to use for unqualified 
+            objects in the SQL, in the form "DB.SCHEMA" 
+            (e.g., "RAW.WIDEWORLDIMPORTERS_WAREHOUSE")
+
+    Returns:
+        Dict[str, Any]: Dictionary containing:
+            - success: Boolean indicating if the query was successful
+            - data: Query result data (rows, columns) if successful
+            - error: Error message if query failed
+            - query_info: Additional query execution information
+
+    Examples:
+        # First, find available connections using search_assets_tool
```

Comment:
Remove the example for fetching connections. Add example on how to fetch the tables etc for write the query on. , `connectionQualifiedName` will be sent as part of that response which can be used in this attribute later on